miner/worker.go

```diff
@@ -1076,16 +1081,89 @@ func (w *worker) prepareWork(genParams *generateParams) (*environment, error) {
+	// remoteTxs before processing them separately and avoid double counting.
 	pending := w.eth.TxPool().Pending(true)
+
+	electroneumTxs := make(map[common.Address]types.Transactions)
+	if w.config.PrioritiseElectroneum {
+
+		//todo: move this into worker or somewhere to save doing this every time we fill a block
+		//testing keys only
+		// private key : 8dd88ba44afa275e0dd9be1d633732242d84ea955e453b28719d17897035b4c2
+		//NB the 0x04 is to tell us that the key is uncompressed 64bytes +1 byte
+		etnSigningPubKey := "04" + "d6e7d6da18e8cfb9cf20bbb7182ac36deb9b77f2c617f3698fc12aebf22ee9b00060775349a2312813818a9c7a0a4b143fddfa36611129daf01c26bc9080d418"
+		byteStringETNKey, err := hex.DecodeString(etnSigningPubKey)
+		if err != nil {
+			fmt.Println("Error decoding hex string:", err)
+			panic("Error decoding etn signing key")
+		}
+
+		// If the first transaction for a given account has the Electroneum Ltd signature present, then all of the tx
+		// for that account do, so we can just pull that account out of pending without further thought
+		for account, txs := range pending {
+			// Extract the nonce - it's a uint64, so we need to convert it to bytes
+			nonceHex := hexutil.EncodeUint64(txs[0].Nonce())
+			nonce, _ := hex.DecodeString(nonceHex[2:]) // [2:] to trim the '0x' prefix
+			// Create signer depending on whether the transaction is protected
+			var signer types.Signer
+			if txs[0].Protected() {
+				signer = types.NewEIP155Signer(w.chainConfig.ChainID)
+			} else {
+				signer = types.HomesteadSigner{}
+			}
+
+			// Extract the sender - the function returns an address, we convert it to bytes
+			var sender []byte
+			if from, err := types.Sender(signer, txs[0]); err != nil {
+				panic(fmt.Sprintf("Failed to get transaction sender: %v", err))
+			} else {
+				sender = from.Bytes()
+			}
+
+			// Concatenate the nonce and sender
+			data := append(nonce, sender...)
+
+			// Generate the hash
+			digestHash := crypto.Keccak256(data)
+
+			//check the last 64 bytes of data to see if there is a myETN signature present.
+			//note that the signature txs[0].Data()[len(txs[0].Data())-64:] is just (r,s) not (r,s,v)
+			if len(txs) > 0 && len(txs[0].Data()) >= 64 { // conditional is >=64 in case we add more to the data field at a later stage for whatever reason
+				r := txs[0].Data()[len(txs[0].Data())-64 : len(txs[0].Data())-32]
+				s := txs[0].Data()[len(txs[0].Data())-32:]
+				if !crypto.ValidateSignatureValues(0, new(big.Int).SetBytes(r), new(big.Int).SetBytes(s), false) { // if r,s aren't suitable then just break and move onto the next tx rather than verifying
+					break
+				}
+				if crypto.VerifySignature(byteStringETNKey, digestHash, txs[0].Data()[len(txs[0].Data())-64:]) {
+					delete(pending, account)
+					electroneumTxs[account] = txs
+				} else {
+					log.Warn("Got a data field signature but couldn't verify: %v", txs[0].Hash())
+				}
+			}
+		}
+	}
+
+	//
```

Comment:
general comment 1: i would put this code above at the txpool level so that we can do:

```
pending := w.eth.TxPool().Pending(true)

if w.config.PrioritiseElectroneum {
  electroneumTxs := w.eth.TxPool().ETNPending(true)
}
```

^ so that we can do something like this at worker level.

--------------------------
general comment 2: 
we need specific global & account slots for etn txs at txpool level, otherwise if txpool slots are full we won't get subsequent etn txs that needs prioritization.

Reply:
GC#1: I agree & this was the todo list already
https://github.com/electroneum/electroneum-sc/blob/53e7c99d64dd2264b0a8168861f271a99946c754/miner/worker.go#L1093
 but I did things this way for now just for expedience. Definitely one to do later.
GC#2: I agree with this. I was looking over it and we currently have:
AccountSlots: 16,
GlobalSlots:  4096 + 1024, // urgent + floating queue capacity with 4:1 ratio
We have 5 second blocks and 1428 standard transactions per block. 
	
So even under genuine and very high load, this is plenty already, but under brief DoS attack, it's possible that some myETN txs fall out of the every* node's pool, so ideally this change does go into the codebase pre release rather than adjusting the myETN ifra's tx lifecycle logic.

Reply:
Decided to just do number 2 now so there is no friction at all.